### PR TITLE
Use react-frame-component for preview pane

### LINKED ISSUE #307: Live preview doesn't work on Firefox/Mac
**- Bug -**

The live preview flashes up and then disappears on Firefox. No errors in browser console.

netlify-admin

### PR DESCRIPTION
**Pull request by Benaiah**
Fixes #307

**- Summary**

Uses `react-frame-component` for preview pane instead of custom iframe code. Fixes #307.

**- Test plan**

Existing tests pass; manually tested that previewing and scrolling work correctly in both Firefox and Chrome.

**- Description for the changelog**

Use react-frame-component for preview pane

### CODE REVIEW

File: src/components/PreviewPane/PreviewPane.js [hunk 1]
```diff
@@ -107,22 +83,25 @@ export default class PreviewPane extends React.Component {
...
 
-    // We need to use this API in order to pass context to the iframe
-    ReactDOM.unstable_renderSubtreeIntoContainer(
-      this,
-      <ScrollSyncPane attachTo={this.iframeBody}>
-        {React.createElement(component, previewProps)}
-      </ScrollSyncPane>
-      , this.previewEl);
-  }
+    const styleEl = <link href={registry.getPreviewStyles()} type="text/css" rel="stylesheet" />;
```
Comment (erquhart) [inline src/components/PreviewPane/PreviewPane.js]: `registry.getPreviewStyles` returns an array - we should be mapping it.

File: src/components/PreviewPane/PreviewPane.js [hunk 2]
```diff
@@ -66,35 +62,15 @@ export default class PreviewPane extends React.Component {
...
-  handleIframeRef = (ref) => {
-    if (ref) {
-      registry.getPreviewStyles().forEach((style) => {
-        const linkEl = document.createElement('link');
-        linkEl.setAttribute('rel', 'stylesheet');
-        linkEl.setAttribute('href', style);
-        ref.contentDocument.head.appendChild(linkEl);
-      });
-
-      const base = document.createElement('base');
```
Comment (erquhart) [inline src/components/PreviewPane/PreviewPane.js]: I believe we'll need this to avoid regression of #172.

**Comment by erquhart** [comment]
@Benaiah update still coming for this?

File: src/components/PreviewPane/PreviewPane.js (hunk 2 above)
Comment (Benaiah) [inline src/components/PreviewPane/PreviewPane.js]: Resolved this by adding it to the `initialContent` prop on `Frame`.

File: src/components/PreviewPane/PreviewPane.js (hunk 1 above)
Comment (Benaiah) [inline src/components/PreviewPane/PreviewPane.js]: Good catch, thanks!

File: src/components/PreviewPane/PreviewPane.js [hunk 3]
```diff
@@ -107,22 +83,35 @@ export default class PreviewPane extends React.Component {
...
+      <ScrollSyncPane attachTo={iFrameDocument.scrollingElement}>
+        {React.createElement(component, previewProps)}
+      </ScrollSyncPane>);
+
+    PreviewContent.contextTypes = {
+      document: PropTypes.any,
+    };
+
+    return (<Frame
+      className={styles.frame}more** crazy stuff
```
Comment (erquhart) [inline src/components/PreviewPane/PreviewPane.js]: lol more** crazy stuff?

Comment (Benaiah) [inline src/components/PreviewPane/PreviewPane.js]: ugh, middle-click paste is not my friend sometimes :stuck_out_tongue:

**Pull request merged**